### DISCUSSION TITLE
Unable to / how to have multiple or nested relationships and query them?

### DISCUSSION THREAD

**Discussion by ttbarnes** [keystonejs/keystone#8245] [Questions]
Hiya, i'm trying to have a nice SQL database structure with a small amount of columns in each table. I'm able to get this structure saved, but it seems impossible to query nested relations.

Here is a simple example of my schema:

```js
// schema.ts
export const lists = {
  Product: {
    fields: {
      createdAt: timestamp(),
      company: relationship({ ref: 'Company' }),
      companyContact: relationship({ ref: 'CompanyContacts', many: true }),
    },
    access: allowAll,
  },
  Company: list({
    fields: {
      product: relationship({ ref: 'Product' }),
      companyContacts: relationship({ ref: 'CompanyContact' }),
      companyName: text(),
    },
    access: allowAll,
  }),
  CompanyContact: list({
    fields: {
      company: relationship({ ref: 'Company' }),
      product: relationship({ ref: 'Product' }),
      firstName: text(),
      lastName: text(),
    },
    access: allowAll,
  }),
};
```

It seems impossible to get `CompanyContact`, no matter how I structure the query. Tried a few combinations, example: 

```js
{
  product(where:{ id: "1234"}) {
    id
    name
    company {
      id
      companyName
      companyContacts {
        id
        firstName
        lastName
      }
    }
    companyContacts {
      id
      firstName
      lastName
    }
  }
}
```

Am I missing something here?

It seems it's possible by doing two graphql queries from the UI, but I want the API to do the heavy lifting instead of making two requests from the UI.

Also seems possible to achieve this by writing a custom schema and manually querying/getting the data via the context DB API. But I really don't want to do this as it's not scalable.

Any help appreciated! 🙏

**Comment by SagnikPradhan** [comment]
Dang, no replies? Did you ever make any development on it?
